src/algorithms/calorimetry/CalorimeterEoverPCut.cc

```diff
@@ -0,0 +1,56 @@
+#include <edm4eic/EDM4eicVersion.h>
+#if EDM4EIC_VERSION_MAJOR >= 8
+
+#include <edm4hep/utils/vector_utils.h>
+#include <cmath>
+#include "CalorimeterEoverPCut.h"
+
+namespace eicrecon {
+
+void CalorimeterEoverPCut::process(const CalorimeterEoverPCut::Input& input,
+                                   const CalorimeterEoverPCut::Output& output) const {
+  const auto& [clusters, assoc_opt] = input;
+  auto& [pid_coll_ptr]              = output;
+
+  if (!assoc_opt) {
+    warning("[E/P Cut] no MC associations, skipping");
+    return;
+  }
```

Comment:
This makes it non-optional, so you should not mark is as `std::optional` in the header file.